pkg/report/sarif.go

```diff
@@ -348,6 +350,14 @@ func clearURI(s string) string {
 	return strings.ReplaceAll(strings.ReplaceAll(s, "\\", "/"), "git::https:/", "")
 }
 
+func toUri(str string) *url.URL {
```

Comment:
Looks like we can return `string` here
```suggestion
func toUri(str string) string {
```

Reply:
It cannot be string if the  sarifData struct changes are kept:

```diff
@@ -61,9 +63,9 @@ type sarifData struct {
        helpMarkdown     string
        resourceClass    types.ResultClass
        severity         string
-       url              string
+       url              *url.URL
        resultIndex      int
-       artifactLocation string
+       artifactLocation *url.URL
```

Does it mean you would prefer sarifData url and artifactLocation not to be strongly typed as url.URL?

Reply:
`helpUri` and `artifactLocation.uri` are string fields. So I thought we could immediately get the required (string) type from the `toUri` output.

But I thought again now, and maybe your solution is better.
It might save us from mistakes later.

So let's use your solution!